src/components/va-back-to-top/test/va-back-to-top.e2e.ts

```diff
@@ -0,0 +1,220 @@
+
+    // We've scrolled a lot - button should be revealed
+    expect(button).toHaveClass('reveal');
+  });
+
+  it('docks when the "dock" is scrolled into view', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`
+    <main>
+    ${pageContent}
+    <va-back-to-top></va-back-to-top>
+    <footer style="height: 100px">The footer</footer>
+    </main>
+    `);
+    const wrapper = await page.find('va-back-to-top >>> div');
+
+    expect(wrapper).not.toHaveClass('docked');
+
+    await page.mouse.wheel({ deltaY: 1400 });
+    await page.waitForChanges();
+
+    // We've scrolled a lot - button should be docked
+    expect(wrapper).toHaveClass('docked');
+  });
+
+  it('stays docked even with really long footers', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`
+    <main>
+    ${pageContent}
+    <va-back-to-top></va-back-to-top>
+    <footer style="height: 2400px">The footer</footer>
+    </main>
+    `);
+    const wrapper = await page.find('va-back-to-top >>> div');
+
+    await page.mouse.wheel({ deltaY: 3000 });
+    await page.waitForChanges();
+
+    // After scrolling this much, the dock is well above the viewport
+    expect(wrapper).toHaveClass('docked');
+  });
+
+  it('goes to top on click', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`
+    <main>
+    ${pageContent}
+    <va-back-to-top></va-back-to-top>
+    </main>
+    `);
+    const button = await page.find('va-back-to-top >>> button');
+    const topHeading = await page.find('h1');
+
+    // Make the button visible
+    await page.mouse.wheel({ deltaY: 1000 });
+    await page.waitForChanges();
+
+    expect(await topHeading.isIntersectingViewport()).toEqual(false);
+
+    button.click();
+    await page.waitForChanges();
+
+    expect(await topHeading.isIntersectingViewport()).toEqual(true);
+  });
+
+  it('passes an axe check', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`
+    <main>
+    ${pageContent}
+    <va-back-to-top></va-back-to-top>
+    </main>
+    `);
+
+    await axeCheck(page);
+  });
+
+  // This is oddly failing on a color contrast error
+  it.skip('passes an axe check when revealed', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`
+    <main>
+    ${pageContent}
+    <va-back-to-top></va-back-to-top>
+    </main>
+    `);
+
+    await page.mouse.wheel({ deltaY: 1000 });
+    await page.waitForChanges();
+
+    await axeCheck(page);
+  });
+
+  // This is oddly failing on a color contrast error
+  it.skip('passes an axe check when docked', async () => {
+    const page = await newE2EPage();
+    await page.setContent(`
+    <main>
+    ${pageContent}
+    <va-back-to-top></va-back-to-top>
+    <footer style="height: 100px">The footer</footer>
+    </main>
+    `);
+
+    await page.mouse.wheel({ deltaY: 1400 });
+    await page.waitForChanges();
+
+    await axeCheck(page);
+  });
```

Comment:
I might be getting false negatives here - I'd like to confirm that if possible. 

Here are the failures when not skipped:

![image](https://user-images.githubusercontent.com/2008881/135291788-cfbaf9d8-bb5c-4b91-b6dc-6ce62a7227ba.png)


Here are some axe tests I did in the browser - I even disabled `formation` on the off chance it was adding some color contrast that the tests weren't getting.

No color contrast errors when revealed:

![image](https://user-images.githubusercontent.com/2008881/135290816-6646303c-a85d-47a5-950e-bffd8e240b78.png)

No color contrast errors when docked:

![image](https://user-images.githubusercontent.com/2008881/135291324-dbc7482b-f239-476f-9b72-242b83afbe00.png)